### PR TITLE
Support absolute paths when rendering a template

### PR DESCRIPTION
**Pull request by FrancescoBorzi**

### CODE REVIEW

**Comment by stof** [comment]
@fabpot we should work on deprecating this pattern IMO, which was allowed due to internal implementation details of the filesystem loader but not intentionally supported (we actually even broke it in the past and reverted some changes to avoid BC breaks for people relying on this behavior).

File: lib/Twig/Loader/Filesystem.php [hunk 1]
```diff
@@ -181,6 +181,12 @@ protected function findTemplate($name, $throw = true)
             return $this->cache[$name];
         }
 
+        if (is_file($name)) {
+            $this->cache[$name] = $name;
+
+            return $name;
```
Comment (stof) [inline lib/Twig/Loader/Filesystem.php]: returning a name without the validation is a no-go. It is a security issue. This logic must go after the validation

**Review by stof — CHANGES_REQUESTED** [review]

File: lib/Twig/Loader/Filesystem.php (hunk 1 above)
Comment (ShinDarth) [inline lib/Twig/Loader/Filesystem.php]: done

**Comment by fabpot** [comment]
Can you add a test?

**Comment by ShinDarth** [comment]
currently I'm on holiday but I can add the test as soon as I'll be back if my proposed change is acceptable

**Comment by stof** [comment]
Btw, this also breaks the promise that Twig does not allow rendering things outside the configured template folders, which could introduce security issues. So I don't like this change.

**Comment by fabpot** [comment]
I'm closing this pull request as I agree with @stof, we don't want to support absolute paths. It worked by chance but was never supposed to be supported.

**Pull request closed without merging**